subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c

```diff
@@ -625,10 +625,26 @@ static void isr_tx_common(void *param,
 
 	/* Get ISO data PDU, not control subevent */
 	if (!pdu) {
-		uint8_t payload_index;
+		uint16_t payload_index;
+
+		if (lll->ptc_curr) {
+			uint8_t ptx_idx = lll->ptc_curr - 1U; /* max. nse 5 bits */
+			uint8_t ptx_payload_idx;
+			uint16_t ptx_group_mult;
+			uint8_t ptx_group_idx;
+
+			/* Calculate group index and multiplier for deriving
+			 * pre-transmission payload index.
+			 */
+			ptx_group_idx = ptx_idx / lll->bn; /* 5 bits */
+			ptx_payload_idx = ptx_idx - ptx_group_idx * lll->bn; /* 8 bits */
+			ptx_group_mult = (ptx_group_idx + 1U) * lll->pto; /* 9 bits */
+			payload_index = ptx_payload_idx + ptx_group_mult * lll->bn; /* 13 bits */
+			LL_ASSERT(payload_index <= UINT8_MAX);
```

Comment:
Something looks a bit here now; you have a comment stating `13 bits` but you also assert that it's less <= 8 bits